libvast/src/system/execution_node.cpp

```diff
@@ -34,10 +34,14 @@ class actor_control_plane : public operator_control_plane {
     : self_{self} {
   }
 
-  auto self() noexcept -> caf::scheduled_actor& override {
+  auto self() noexcept -> system::execution_node_actor::base& override {
     return self_;
   }
 
+  auto node() noexcept -> system::node_actor override {
```

Comment:
This should probably return an `expected` instead of just returning a default-initialized node actor for local actors. While the latter may be okay, there are just too many CAF APIs that do not gracefully handle nullptr destination handles.